crates/rattler/src/install/mod.rs

```diff
@@ -876,4 +905,40 @@ mod test {
 
         insta::assert_yaml_snapshot!(paths);
     }
+
+    #[tracing_test::traced_test]
+    #[tokio::test]
+    async fn test_link_package_from_archive() {
+        let packages = vec![(
+            "https://conda.anaconda.org/conda-forge/win-64/ruff-0.0.171-py310h298983d_0.conda",
+            "25c755b97189ee066576b4ae3999d5e7ff4406d236b984742194e63941838dcd"
+        ),
+        (
+            "https://conda.anaconda.org/conda-forge/linux-aarch64/bzip2-1.0.8-hf897c2e_4.tar.bz2",
+            "3aeb6ab92aa0351722497b2d2a735dc20921cf6c60d9196c04b7a2b9ece198d2"
+        )];
+
+        for (package_url, sha256) in packages {
+            let environment_dir = tempfile::TempDir::new().unwrap();
+
+            let package_path =
+                tools::download_and_cache_file_async(package_url.parse().unwrap(), sha256)
+                    .await
+                    .unwrap();
+
+            let install_driver = InstallDriver::default();
+
+            // Link the package
+            let paths = link_package_from_archive(
+                &package_path,
+                environment_dir.path(),
+                &install_driver,
+                InstallOptions::default(),
+            )
+            .await
+            .unwrap();
+
+            insta::assert_yaml_snapshot!(paths);
```

Comment:
Can you give the snapshot a name? This guarentees that the snapshot is properly linked to the loop iteration. Otherwise we might get flaky tests later.